lib/Mail/Milter/Authentication/Handler.pm

```diff
@@ -2517,7 +2517,9 @@ sub add_headers {
         $header .= '; none';
     }
 
-    $self->prepend_header( 'Authentication-Results', $header );
+    if ( @auth_headers || !$config->{'hide_none'} ) {
```

Comment:
The flow here feels a little clumsy, I would move the prepend_header up into the if @auth_headers block, add an 'elsif (!$config->{hide_none}) {' block which adds the none result, and a final else block which is a nop with a comment.

Reply:
SGTM, I've updated the patch.